### PR TITLE
update recv to use correct values on FreeBSD

### PR DESCRIPTION
**Pull request by 180watt**
It's rather unfortunate that Rust's libc doesn't know that FreeBSD has POLLRDHUP, but in the mean time we can add the value ourselves. We also need to use a uint for `nfds_t`. It seems that illumos also has POLLRDHUP, so this implementation could work there too. (Testing welcome.) I'm a novice when it comes to Rust, so any suggestions to this patch is very welcomed.

P.S. I failed to mention this, but in its current state (pre-patch), ipc-channel fails to build on FreeBSD

### CODE REVIEW

**Comment by wusyong** [comment]
Can you either change only the parts that are affected, like `POLLRDHUP` and `poll,` or add a comment for the method to describe the difference between Linux and FreeBSD?

**Comment by 180watt** [comment]
Updated my commit to be more "surgical." Thanks for the suggestions.

File: src/platform/unix/mod.rs [hunk 1]
```diff
@@ -1162,18 +1162,36 @@ impl UnixCmsg {
...
+                #[cfg(target_os = "linux")]
                 let events = libc::POLLIN | libc::POLLPRI | libc::POLLRDHUP;
+
+                // It's rather unfortunate that Rust's libc doesn't know that
+                // FreeBSD has POLLRDHUP, but in the mean time we can add
+                // the value ourselves.
+                // https://cgit.freebsd.org/src/tree/sys/sys/poll.h#n72
+                #[cfg(target_os = "freebsd")]
+                let events = libc::POLLIN | libc::POLLPRI | 0x4000;
+
```
Comment (jdm) [inline src/platform/unix/mod.rs:1174]: Could we instead add a POLLRDHUP constant that is conditionally defined, rather than the implementation code here?

Comment (180watt) [reply src/platform/unix/mod.rs:1174]: Most certainly.

**Review by wusyong — APPROVED** [review]

File: src/platform/unix/mod.rs [hunk 2]
```diff
@@ -1162,18 +1162,36 @@ impl UnixCmsg {
...
                     fd.as_mut_ptr(),
                     fd.len() as libc::c_ulong,
                     duration.as_millis().try_into().unwrap_or(-1),
                 );
 
+		// FreeBSD's nfds_t is an unsigned integer.
+		#[cfg(target_os = "freebsd")]
+                let result = libc::poll(
+                    fd.as_mut_ptr(),
+                    fd.len() as libc::c_uint,
```
Comment (jdm) [inline src/platform/unix/mod.rs:1191]: We should be able to use `as _` and avoid any platform-specific code.

Comment (180watt) [reply src/platform/unix/mod.rs]: Fixed up.

**Comment by 180watt** [comment]
If this patch can also cause a bump in minor version, that would be greatly appreciated.

**Review by jdm — APPROVED** [review]
Thanks!

**Comment by 180watt** [comment]
I have missed a critical `c` in the code, and we are now trying to do `lib::c_short`. Of course this had to be in the path that I _don't_ build, so I didn't see the error. I am **very** sorry about that. I am also currently away from my computer and can not readily push something. Do you mind either reverting or pushing a fix?

**Pull request merged**